echopype/calibrate/cal_params.py

```diff
@@ -202,61 +296,211 @@ def get_vend_cal_params_power(beam: xr.Dataset, vend: xr.Dataset, param: str) ->
+def get_cal_params_AZFP(beam: xr.DataArray, vend: xr.DataArray, user_dict: dict) -> dict:
     """
-    Get broadband gain or angle factor for calibrating complex samples.
-
-    Interpolate gain or angle factor in the Vendor_specific group to the center frequency
-    of each ping for BB mode samples if nominal frequency is within the calibrated frequency range
+    Get cal params using user inputs or values from data file.
 
     Parameters
     ----------
-    vend : xr.Dataset
-        Vendor_specific group in an EchoData object
-    varname : str
-        the desired parameter
-    freq_center : xr.DataArray
-        center frequency of the transmit BB signal
-    cal_params_CW : dict
-        a dictionary storing CW calibration parameters as xr.DataArray
+    echodata : EchoData
+        An EchoData object containing data to be calibrated
+    user_cal_dict : dict
+        A dictionary containing user-defined calibration parameters.
+        The user-defined calibration parameters will overwrite values in the data file.
 
     Returns
     -------
-    An xr.DataArray containing the interpolated gain or angle factor
+    A dict containing the calibration parameters for the AZFP echosounder
     """
-    param = []
-    for ch_id in freq_center["channel"]:
-        # if frequency-dependent gain/angle factor exists in Vendor group,
-        # interpolate at center frequency
-        if "cal_channel_id" in vend.coords and ch_id.data in vend["cal_channel_id"]:
-            param_temp = (
-                vend[PARAM_VEND[varname]]
-                .sel(cal_channel_id=ch_id)
-                .interp(cal_frequency=freq_center.sel(channel=ch_id))
-                .drop(["cal_channel_id", "cal_frequency"])
-                .expand_dims("channel")
-            )
-        # if no frequency-dependent gain/angle factor exists, use CW gain or default value
+    # Use sanitized user dict as blueprint
+    # out_dict contains only and all of the allowable cal params
+    out_dict = sanitize_user_cal_dict(
+        user_dict=user_dict, channel=beam["channel"], sonar_type="AZFP"
+    )
+
+    # Only fill in params that are None
+    for p, v in out_dict.items():
+        if v is None:
+            # Params from Sonar/Beam_group1
+            if p == "equivalent_beam_angle":
+                # equivalent_beam_angle has dims: channel, ping_time, beam --> only need channel
+                out_dict[p] = beam[p].isel(ping_time=0, beam=0).drop(["ping_time", "beam"])
+
+            # Params from Vendor_specific group
+            elif p in ["EL", "DS", "TVR", "VTX", "Sv_offset"]:
+                out_dict[p] = vend[p]  # these params only have the channel dimension
+
+    return out_dict
+
+
+def get_cal_params_EK(
+    waveform_mode: str,
+    freq_center: xr.DataArray,
+    beam: xr.Dataset,
+    vend: xr.Dataset,
+    user_dict: Dict[str, Union[int, float, xr.DataArray]],
+    default_dict: Dict[str, Union[int, float]] = EK80_DEFAULT_PARAMS,
```

Comment:
Maybe can rename `default_dict` to `default_params` to make it more sense?

Reply:
Good point, done!